src/adaptation/massmatrix.jl

```diff
@@ -102,20 +102,50 @@ function WelfordVar(sz::Union{Tuple{Int},Tuple{Int,Int}}; kwargs...)
     return WelfordVar{Float64}(sz; kwargs...)
 end
 
-function Base.resize!(wv::WelfordVar, θ::AbstractVecOrMat{T}) where {T<:AbstractFloat}
-    if size(θ) != size(wv.var)
+function Base.resize!(wv::WelfordVar, θ::AbstractMatrix{T}) where {T<:AbstractFloat}
+    size_θ = size(θ)
+    if size_θ != size(wv.var)
         @assert wv.n == 0 "Cannot resize a var estimator when it contains samples."
-        wv.μ = zeros(T, size(θ))
-        wv.M = zeros(T, size(θ))
-        wv.δ = zeros(T, size(θ))
-        wv.var = ones(T, size(θ))
+        wv.μ = zeros(T, size_θ)
+        wv.M = zeros(T, size_θ)
+        wv.δ = zeros(T, size_θ)
+        wv.var = ones(T, size_θ)
     end
 end
 
+function Base.resize!(wv::WelfordVar, θ::AbstractVector{T}) where {T<:AbstractFloat}
+    n = length(θ)
+    if n != size(wv.var, 1)
+        @assert wv.n == 0 "Cannot resize a var estimator when it contains samples."
+        __resize_fill!(wv.μ, n, T(0))
+        __resize_fill!(wv.M, n, T(0))
+        __resize_fill!(wv.δ, n, T(0))
+        __resize_fill!(wv.var, n, T(1))
+    end
+end
+
+"""
+    __resize_fill!(x::AbstractVector, new_length::Int, v::T)
+
+Resize `x` according to `new_length`:
+ - If `new_length` is bigger than the current length of `x`, expand `x` to `new_length` and fill `x` with `v`.
+ - If `new_length` is smaller than the current length of `x`, resize `x` to `new_length` and fill `x` with `v`.
+"""
+function __resize_fill!(x::AbstractVector, new_length::Int, v::T) where {T<:Number}
+    length_diff = new_length - length(x)
+    if length_diff > 0
+        append!(x, Vector{T}(undef, length_diff))
+    else
+        resize!(x, new_length)
+    end
```

Comment:
Both branches seem to be just
```suggestion
    resize!(x, new_length)
```
? As mentioned above, in that case `fill!(resize!(..), ..)` would be simpler than a separate function IMO.

Reply:
Good idea! Just changed to simpler resize and fill operations